src/bci_build/package/__init__.py

```diff
@@ -402,12 +402,43 @@ def build_name(self) -> str | None:
 
     @property
     def build_version(self) -> str | None:
+        """Define the BuildVersion that is used for determining which container build
+        is newer than the other in the build service."""
+        # KIWI used to require an at least 3 component version X.Y.Z. For SLE, we set
+        # X.Y to MAJOR.MINOR and set Z to 0 for OsContainers. Derived
+        # containers inhert the base build_version X.Y and append a suffix .Z.ZZ
+
+        # It is important that the behavior for OsContainers is identical
+        # between KIWI and Dockerfile builds so that we can switch between these
+        # types.
+
+        if self.os_version.is_tumbleweed and isinstance(self, OsContainer):
+            return f"{str(datetime.datetime.now().year)}.0.0"
+        # TODO: also set it for non-kiwi type (historically we haven't done so)
+        if self.os_version.is_tumbleweed and self.build_recipe_type == BuildType.KIWI:
+            return f"{str(datetime.datetime.now().year)}.0"
+        if self.os_version.is_sle15 and isinstance(self, OsContainer):
+            return f"15.{int(self.os_version.value)}.0"
         if self.os_version.is_sle15:
             return f"15.{int(self.os_version.value)}"
+        if self.os_version.is_sl16 and isinstance(self, OsContainer):
+            return f"{str(self.os_version.value)}.0"
         if self.os_version.is_sl16:
             return str(self.os_version.value)
         return None
 
+    @property
+    def kiwi_version(self) -> str | None:
+        """Return a BuildVersion that is compatible with the requirements that KIWI imposes.
+        https://osinside.github.io/kiwi/image_description/elements.html#preferences-version
+
+        It a version strictly in the format X.Y.Z.
+        """
+        build_ver: str | None = self.build_version
+        if build_ver:
+            return ".".join(build_ver.split(".")[:3])
```

Comment:
FWICT the `[:3]` will only truncate, not extend, so if build_ver is just X or X.Y, this won't add `.0` or so

Reply:
yes, but build_version will be always 3 or 4 components (oscontainers or develcontainers). we can add tests for this.

Reply:
Apparently trivy-image has `<version>2025.0</version>` now